tools/umm.py

```diff
@@ -76,9 +76,19 @@ def base_name(x):
     return x[1]
 
 # This assumes that membership is a DAG which is the case in C
-# We could memoize, doesn't seem worth it ...
 
 
+def memoize(func):
```

Comment:
So is this worth it now or not?  And when we add this, the comment "# This assumes that membership is a DAG which is the case in C" should be moved down where `paths_to_type` is declared.

Reply:
My initial thought was it doesn't hurt, but you are right, memoization is complete overkill here, the runtime of the entire program is already insignificant, and it just adds unneeded complexity. So, I'll remove it and the comment.